content/blog/2019-09-11-migrate-from-helm-v2-to-helm-v3.md

```diff
@@ -0,0 +1,223 @@
+  -t, --tiller-ns string         namespace of Tiller (default "kube-system")
+      --tiller-out-cluster       when  Tiller is not running in the cluster e.g. Tillerless
+```
+
+Nice, the plugin even supports the [Tillerless Helm v2](https://github.com/rimusz/helm-tiller).
+
+Let's check out for Helm v2 releases and pick one to test out the migration:
+
+```
+$ helm list
+
+NAME    	REVISION	UPDATED                 	STATUS  	CHART           	APP VERSION	NAMESPACE
+postgres	1       	Wed Sep 11 14:52:32 2019	DEPLOYED	postgresql-6.3.5	11.5.0     	postgres
+redis   	1       	Wed Sep 11 14:52:57 2019	DEPLOYED	redis-9.1.7     	5.0.5      	redis
+```
+
+The safest way of course to start with `--dry-run` flag:
+
+```
+$ helm3 2to3 convert --dry-run postgres
+NOTE: This is in dry-run mode, the following actions will not be executed.
+Run without --dry-run to take the actions described below:
+
+Release "postgres" will be converted from Helm 2 to Helm 3.
+[Helm 3] Release "postgres" will be created.
+[Helm 3] ReleaseVersion "postgres.v1" will be created.
+```
+
+Now, let's run the actual migration:
+
+```
+$ helm3 2to3 convert postgres
+Release "postgres" will be converted from Helm 2 to Helm 3.
+[Helm 3] Release "postgres" will be created.
+[Helm 3] ReleaseVersion "postgres.v1" will be created.
+[Helm 3] ReleaseVersion "postgres.v1" created.
+[Helm 3] Release "postgres" created.
+Release "postgres" was converted successfully from Helm 2 to Helm 3. Note: the v2 releases still remain and should be removed to avoid conflicts with the migrated v3 releases.
+```
+
+Check out whether it was succesful:
+
+```
+$ helm list
+NAME    	REVISION	UPDATED                 	STATUS  	CHART           	APP VERSION	NAMESPACE
+postgres	1       	Wed Sep 11 14:52:32 2019	DEPLOYED	postgresql-6.3.5	11.5.0     	postgres
+redis   	1       	Wed Sep 11 14:52:57 2019	DEPLOYED	redis-9.1.7     	5.0.5      	redis
+
+$ helm3 list
+NAME    	NAMESPACE	REVISION	UPDATED                             	STATUS  	CHART
+postgres	postgres 	1       	2019-09-11 12:52:32.529413 +0000 UTC	deployed	postgresql-6.3.5
+```
+
+**Note:** As we did not specify `--delete-v2-releases` flag Helm v2 `postgres` release information was left in-tact, it can be deleted with `kubectl` later one.
+
+When are you ready to move all your releases, you can automate it with running `helm list` in a loop and applying `helm3 2to3 convert RELEASE` for each Helm v2 release.
+
+If you are using Tillerless Helm v2, just add `--tiller-out-cluster` to migrate the release:
+
+```
+$ helm3 2to3 convert postgres --tiller-out-cluster
+```
+
+Very cool and simple, right :-)
+
+## Clean up of Helm v2 data
+The last step is cleaning up the old data. While this is not required, we strongly recommend it.
+
+To clean up Helm v2 data:
+
+- Delete home folder `.helm`
+- Delete Tiller releases if you haven't used `--delete-v2-releases` flag
+- Uninstall Tiller
+
+### Happy Helm v3 sailing
```

Comment:
"Happy Helm v3 sailing" should not be a section heading. Would bold work instead?